src/vos/vos_obj_index.c

```diff
@@ -233,8 +233,12 @@ vos_oi_find_alloc(struct vos_container *cont, daos_unit_oid_t oid,
 		DP_UOID(oid));
 
 	rc = vos_oi_find(cont, oid, &obj, ts_set);
-	if (rc == 0)
+	if (rc == 0) {
+		/* upgrading case, set it to current epoch */
+		if (obj->vo_max_write == 0)
```

Comment:
Does this always work?   This code only executes if we hold the object which we don't do during iteration

Reply:
You are right, after some more testing, i think codes should be fixed in oi_iter_fetch() directly where ie_last_update is inited.